Use stdlib python logging for pass manager logging

mtreinish opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add it in the CHANGELOG file under Unreleased section.
⚠️ If your pull request fixes an open issue, please link to the issue.

✅ I have added the tests to cover my changes.
✅ I have updated the documentation accordingly.
✅ I have read the CONTRIBUTING document.
-->

### Summary

This commit reworks the pass manager logger to use python stdlib logging
instead of attributes in the Passmanager class. This will enable users
to use standard logging mechanism and capture to access debug messages
from the pass manager.

### Details and comments

kdk commented:
AFAICT, the minimal code a user would need to write to enable the logger would be:
```
import logging; import sys
logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
```

What do you think about adding a top-level kwarg to transpile like `transpile(qc, log_level='DEBUG')` that would enable logging to stdout at a given level for the duration of the call to `transpile`? Also, maybe we can log pass order and timing at INFO and leave DEBUG for more detailed info.

mtreinish commented:
Well you shouldn't need to the `sys.stdout` out bit unless stdout is a hard requirement. In the default case the default handler uses stderr for logging output which will show up for most people. But yeah, you will need to specify the logging level somewhere to tell the python logging you want to use a more detailed log level than warning. Sure I can add a flag to transpile to set the log level to something else, that makes sense to me.

As for the specific log levels and log messages I'm not tied to anything in particular. If you have suggestions on what level and the msg format to be I'll gladly change it. I wasn't tied to the format I used, it was more to just have something as a starting point.

mtreinish commented:
@kdk Ok, I've added the `transpile()` parameter for specifying a  log_level with a default of INFO (although I should probably change that to be WARNING instead). I also changed the level of the message emitted from debug to info. But, I'm not sure we want to use the format and level I put in here.
